src/libs/Localize/index.ts

```diff
@@ -82,39 +79,65 @@ const translationCache = new Map<ValueOf<typeof CONST.LOCALES>, Map<TranslationP
  * available in the given locale. If it is, it translates the
  * phrase and stores the translated value in the cache and returns
  * the translated value.
- *
- * @param language
- * @param phraseKey
- * @param fallbackLanguage
- * @param phraseParameters
  */
-function getTranslatedPhrase<TKey extends TranslationPaths>(
+function getTranslatedPhrase<TPath extends TranslationPaths>(
     language: 'en' | 'es' | 'es-ES',
-    phraseKey: TKey,
-    fallbackLanguage: 'en' | 'es' | null = null,
-    ...phraseParameters: PhraseParameters<Phrase<TKey>>
+    path: TPath,
+    fallbackLanguage: 'en' | 'es' | null,
+    ...parameters: TranslationParameters<TPath>
 ): string | null {
     // Get the cache for the above locale
     const cacheForLocale = translationCache.get(language);
 
     // Directly access and assign the translated value from the cache, instead of
     // going through map.has() and map.get() to avoid multiple lookups.
-    const valueFromCache = cacheForLocale?.get(phraseKey);
+    const valueFromCache = cacheForLocale?.get(path);
 
     // If the phrase is already translated, return the translated value
     if (valueFromCache) {
         return valueFromCache;
     }
 
-    const translatedPhrase = translations?.[language]?.[phraseKey] as Phrase<TKey>;
+    const translatedPhrase = translations?.[language]?.[path];
 
     if (translatedPhrase) {
         if (typeof translatedPhrase === 'function') {
-            return translatedPhrase(...phraseParameters);
+            /**
+             *
+             * is Plain object is for checking if the phraseTranslated output
+             * is an object then further check if it include the count param or not
+             * OR before checking the plain object output, we can check if we have the count
+             * param in parameters
+             *
+             */
+            const translateFunction = translatedPhrase as unknown as (...parameters: TranslationParameters<TPath>) => string | PluralForm;
+            const translateResult = translateFunction(...parameters);
+
+            if (typeof translateResult === 'string') {
+                return translateResult;
+            }
+
+            const phraseObject = parameters[0];
+            if (phraseObject && typeof phraseObject === 'object' && 'count' in phraseObject && typeof phraseObject.count === 'number') {
+                const pluralRule = new Intl.PluralRules(language).select(phraseObject.count);
+
+                const pluralResult = pluralRule in translateResult && translateResult[pluralRule];
+                if (pluralResult) {
+                    if (typeof pluralResult === 'string') {
+                        return pluralResult;
+                    }
+
+                    return pluralResult(phraseObject.count);
+                }
+
+                return translateResult.other(phraseObject.count);
+            }
+
+            throw new Error(`Invalid plural form for '${path}'`);
```

Comment:
Throw early (ie: invert [this if](https://github.com/Expensify/App/pull/48229/files#diff-fe6e9abcee003c3390e38c0210d32ff889767cfab2c007837ed64c02147162e1R121) condition and throw)

Reply:
What all previous conditions? There's only one of them.
We prefer early throw/return in general, as it makes the code easier to read and follow